src/v2/components/Billing/components/BillingForm/index.tsx

```diff
@@ -30,93 +27,172 @@ import PlanChanges from 'v2/components/Billing/components/PlanChanges'
 import CancellationNotice from 'v2/components/Billing/components/CancellationNotice'
 import CancelPremium from 'v2/components/Billing/components/CancelPremium'
 import StatusOverlay from 'v2/components/Billing/components/StatusOverlay'
-
-const OPERATIONS = {
-  CHANGE_PLAN_ID: 'CHANGE_PLAN_ID',
-  CANCEL_PREMIUM_SUBSCRIPTION: 'CANCEL_PREMIUM_SUBSCRIPTION',
-  DOWNGRADE_TO_LIFETIME: 'DOWNGRADE_TO_LIFETIME',
-  APPLY_COUPON_CODE: 'APPLY_COUPON_CODE',
-}
+import {
+  SubscribeToPremium as SubscribeToPremiumType,
+  SubscribeToPremiumVariables,
+} from '__generated__/SubscribeToPremium'
+import { CancelPremiumSubscription as CancelPremiumSubscriptionType } from '__generated__/CancelPremiumSubscription'
+import {
+  ApplyCouponToSubscription as ApplyCouponToSubscriptionType,
+  ApplyCouponToSubscriptionVariables,
+} from '__generated__/ApplyCouponToSubscription'
+import { DowngradeToLifetime } from '__generated__/DowngradeToLifetime'
+import { Billing as Me } from '__generated__/Billing'
+import useMergeState from 'v2/hooks/useMergeState'
+import { SupportedPlanEnum } from '__generated__/globalTypes'
+import {
+  SubscribeToPremiumWithoutToken,
+  SubscribeToPremiumWithoutTokenVariables,
+} from '__generated__/SubscribeToPremiumWithoutToken'
+
+type OperationsEnum =
+  | 'CHANGE_PLAN_ID'
+  | 'CANCEL_PREMIUM_SUBSCRIPTION'
+  | 'DOWNGRADE_TO_LIFETIME'
+  | 'APPLY_COUPON_CODE'
 
 interface BillingFormProps {
-  applyCouponToSubscription: any
-  cancelPremiumSubscription: any
-  downgradeToLifetimeSubscription: any
-  me: any
+  me: Me
   onSuccess?: any
   plan_id?: string
-  subscribeToPremium: any // FIXME: Type
 }
 
 interface BillingFormState {
-  mode: string
-  errorMessage: string
-  operations: any[]
-  plan_id: string
-  coupon_code: string
+  mode:
+    | 'resting'
+    | 'error'
+    | 'processing'
+    | 'canceled'
+    | 'subscribed'
+    | 'card_changed'
+  operations: OperationsEnum[]
+  planId?: SupportedPlanEnum | 'basic' | 'lifetime'
```

Comment:
This seems weird? Shouldn't these be encapsulated in that enum as well?

Reply:
Yeah it's a bit weird, but those are not supported plans (but they are values that could potentially be set).